modin/backends/base/doc_utils.py

```diff
@@ -658,6 +658,9 @@ def doc_groupby_method(result, refer_to, action=None):
     drop : bool, default: False
         If `by` is a QueryCompiler indicates whether or not by-data came
         from the `self`.
+    selection : label or list of labels, optional
```

Comment:
This is like doing a `mask` before the groupby, right? Trying to understand.

Reply:
Not exactly, it's more like doing a `mask` on the data, that's required to be aggregated, so it's like `mask` of the raw frame's data after grouping:
<details><summary>Code</summary>

```python
>>> df
   a  b  c
0  1  5  3
1  1  4  4
2  2  3  5
>>> df[["a", "b"]].groupby("a").sum()
   b
a
1  9
2  3
>>> df.groupby("a")[["a", "b"]].sum()
   a  b
a
1  2  9
2  2  3
```

</details>

We can't achieve an effect of aggregating 'by' columns only doing a mask before groupby, unless we'll detach 'by' data from the frame:
<details><summary>Code</summary>

```python
>>> df[["a", "b"]].groupby(df["a"]).sum()
   a  b
a
1  2  9
2  2  3
```

</details>

For now, there's no convention on what backend's groupby obtains as 'by', in some cases, it's detached columns (as query compiler of the 'by' columns), in other cases, it's a list of string column names.

It's not obvious which way of passing 'by' was added by mistake and so which one should be avoided, most of our implementation implies that 'by' is a query compiler, supposing that 'by' as query compiler is the true way, however #2744 notation (as I thought this PR should show an idealistic API that we want to follow) it [forces us to always use columns names](https://github.com/RehanSD/modin/blob/d31e9640c205489716f9bcb61d61ee82e79bfb44/modin/engines/modin_dataframe.py#L257-L258) as 'by's.

To be able to work properly with both of 'by' types, we must pass extra information about actual columns that have to be aggregated (`selection` parameter).

BTW, our current groupby implementation for pandas backend always translates "detached" 'by' into the "attached" ones ([[1]](https://github.com/modin-project/modin/blob/c4a7355e84c3eb23891555237f29c340b864a0d9/modin/backends/pandas/query_compiler.py#L2719-L2723) [[2]](https://github.com/modin-project/modin/blob/c4a7355e84c3eb23891555237f29c340b864a0d9/modin/data_management/functions/groupby_function.py#L124-L133)) so we will always need to generate some kind of meta-info about actual selection (if we remove `selection` backend parameter, we still need this info, so it would be computed anyway inside backend's groupby).

Reply:
In the case of doing a copy, we lose information about where this column came from (whether it originally was from the 'self' frame or it was an external Series). In some cases, GroupBy results depend on the origin of the 'by' data. Anyway, I can try this and see how it goes.

As for now, I decided to detach and move a small piece of this PR that's fixing the actual #3252 issue into a separate PR (#3463), so we can merge the fix until the 0.11 release. 

As for this PR (#3298), it's more about supporting `groupby.__getitem__` in a proper way, it's much more general than the bug fix, so I convert it to draft for now and will try to get rid of adding 'selection' as a new backend's API.